libraries/exoplayer/src/main/java/androidx/media3/exoplayer/drm/KeyRequestInfo.java

```diff
@@ -0,0 +1,66 @@
+package androidx.media3.exoplayer.drm;
+
+import androidx.annotation.Nullable;
+import androidx.media3.common.DrmInitData.SchemeData;
+import androidx.media3.common.util.Assertions;
+import androidx.media3.exoplayer.source.LoadEventInfo;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+
+/**
+ * Encapsulates info for the sequence of load requests ({@link LoadEventInfo}, which were required
+ * to complete loading a DRM key
+ */
+public class KeyRequestInfo {
+
+  public static class Builder {
+    @MonotonicNonNull private LoadEventInfo loadEventInfo;
+    private final List<LoadEventInfo> retriedLoadRequests;
+    @Nullable private final List<SchemeData> schemeDatas;
+
```

Comment:
Clean up the lint, `schemeDatas` is nullable for offline keys (we may want to add an additional identifying member for this case).

`loadEventInfo` is non-null after the main request is indemnified (which must be before `build()`